Add Makefile, configuration files and scripts

razvand opened the pull request:
Update the README with instructions on setting up, configuring, building and running the Nginx app with Unikraft. As part of it, add Makefile files, configuration files and scripts; they are referenced in the `README.md` file.

Current instructions cover the use of QEMU/KVM platform (both on x86_64 and on AArch64) and GCC. Firecraker/KVM, Xen, Linux platforms and Clang are not yet documented.

razvand committed:
Add Makefile, configuration files and scripts

Update the README with instructions on setting up, configuring, building
and running the Nginx app with Unikraft. As part of it, add Makefile
files, configuration files and scripts; they are referenced in the
`README.md` file.

Current instructions cover the use of QEMU/KVM platform (both on x86_64 and
on AArch64) and GCC. Firecraker/KVM, Xen, Linux platforms and Clang are
not yet documented.

Signed-off-by: Razvan Deaconescu <razvand@unikraft.io>

StefanJum commented on config-qemu-x86_x64:
Rename this to `config-qemu-x86_64` (not `x86-x64`).

StefanJum commented on README.md:
````diff
@@ -1,93 +1,202 @@
+     It should print something like this:
+
+     ```console
+     $ tree --charset=ascii -F -L 2 .unikraft/
````
```suggestion
     $ tree -F -L 2 .unikraft/
```

StefanJum commented on README.md:
````diff
@@ -1,93 +1,202 @@
+
+  1. While inside the `.unikraft/` directory, clone the library repositories in the `libs/` directory:
+
+     ```console
````
```suggestion
     ```console
     $ cd libs/
```

StefanJum commented on README.md:
````diff
@@ -1,93 +1,202 @@
+  1. Get back to the application directory:
+
+     ```console
+     $ cd ../
````
```suggestion
     $ cd ../../
```

StefanJum commented on README.md:
````diff
@@ -1,93 +1,202 @@
+Use the `config-qemu-x86_64` configuration file together with `make defconfig` to create the configuration file:
+
+```console
+$ UK_DEFCONFIG=$(pwd)/config-qemu-x84_64 make defconfig
````
```suggestion
$ UK_DEFCONFIG=$(pwd)/config-qemu-x86_64 make defconfig
```

StefanJum reviewed (changes requested):
@razvand first round

StefanJum commented on run-qemu-aarch64.sh:
```diff
@@ -0,0 +1,15 @@
+    -netdev bridge,id=en0,br=virbr0 -device virtio-net-pci,netdev=en0 \
+    -append "netdev.ipv4_addr=172.44.0.2 netdev.ipv4_gw_addr=172.44.0.1 netdev.ipv4_subnet_mask=255.255.255.0 --" \
+    -kernel build/nginx_qemu-arm64 -nographic \
+    -machine virt -cpu cortex-a57
```
```suggestion
    -machine virt -cpu max
```

razvand committed:
Add .gitignore file

Ignore files and folder that are output from the set up, configure,
build or run steps.

Signed-off-by: Razvan Deaconescu <razvand@unikraft.io>

StefanJum commented on README.md:
```diff
@@ -1,93 +1,307 @@
-    inet6 fe80::383e:88ff:fee6:a1e4/64 scope link
-       valid_lft forever preferred_lft forever
+
+At the end of the build command, the `app-nginx_qemu-x86_64` unikernel image is generated.
```
```suggestion
At the end of the build command, the `nginx_qemu-x86_64` unikernel image is generated.
```

StefanJum commented on README.md:
```diff
@@ -1,93 +1,307 @@
-en0: Interface is up
+$ make CROSS_COMPILE=~/packages/gcc-arm-11.2-2022.02-x86_64-aarch64-none-linux-gnu/bin/aarch64-none-linux-gnu- -j $(nproc)
 [...]
+  LD      app-nginx_qemu-arm64.dbg
+  UKBI    app-nginx_qemu-arm64.dbg.bootinfo
+  SCSTRIP app-nginx_qemu-arm64
+  GZ      app-nginx_qemu-arm64.gz
```
```suggestion
  LD      nginx_qemu-arm64.dbg
  UKBI    nginx_qemu-arm64.dbg.bootinfo
  SCSTRIP nginx_qemu-arm64
  GZ      nginx_qemu-arm64.gz
```

StefanJum commented on README.md:
```diff
@@ -1,93 +1,307 @@
+that is, DO NOT fill the command name: `aarch64-none-linux-gnu-gcc`;
+simply end it with `aarch64-none-linux-gnu-`.
+
+Similarly to x86_64, at the end of the build command, the `app-nginx_qemu-arm64` unikernel image is generated.
```
```suggestion
Similarly to x86_64, at the end of the build command, the `nginx_qemu-arm64` unikernel image is generated.
```

StefanJum commented on fs0/nginx/logs/.gitignore:
What is with this file?

razvand replied:
Placeholder to still keep "alive" the `nginx/logs/` directory.

StefanJum commented on README.md:
```diff
@@ -1,93 +1,312 @@
+### QEMU x86_64
+
+Building for QEMU x86_64 assumes you did the QEMU x86_64 configuration step above.
+Build the Unikraft Nginx image for QEMU AArch64 by using the command below:
```
```suggestion
Build the Unikraft Nginx image for QEMU x86_64 by using the command below:
```

StefanJum reviewed (changes requested):
Looks fine besides the comment

StefanJum reviewed (approved):
All good, thanks.

Reviewed-by: Stefan Jumarea <stefanjumarea02@gmail.com>
Approved-by: Stefan Jumarea <stefanjumarea02@gmail.com>

John-Ted reviewed (approved):
All good!

Reviewed-by: Ioan-Teodor Teugea ioan_teodor.teugea@stud.acs.upb.ro
